Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/MainActivity.java

```diff
@@ -205,60 +211,45 @@ public void onFailure(@NonNull Exception e) {
+      Long offset = 1000L / fps;
+      for (int i = 0; i < resultsOCR.size(); i++) {
+        deltaOCR.add(recordStartTime + (i * offset));
       }
     } catch (Exception e) {
       Log.d("Rokus Logs:", "Parsing OCR failed with: " + e.getMessage());
     }
   }
 
-  // Parse timestamps received from server. Last 12 characters of time is important here.
+  // Parse timestamps received from server.
   private void parseServer() {
-    Integer length = serverCache.get(0).length() - 12;
     for (int i = 0; i < serverCache.size(); i++) {
-      String tmp = serverCache.get(i).substring(length);
-      deltaServer.add(convertStringTimeToInt(tmp));
+      deltaServer.add(Long.valueOf(serverCache.get(i)));
       Log.d(tag, "ParseServer:" + deltaServer.get(deltaServer.size() - 1));
     }
   }
 
-  // Converts time in String format hh:mm:ss.MsMsMs (Eg: 08:47:56.637) to Integer millisecondsy
-  private Integer convertStringTimeToInt(@NotNull String currentTime) {
-    Integer result = 0;
-    result += (Integer.parseInt(currentTime.substring(0, 2))) * 3600000;
-    result += (Integer.parseInt(currentTime.substring(3, 5))) * 60000;
-    result += (Integer.parseInt(currentTime.substring(6, 8))) * 1000;
-    result += (Integer.parseInt(currentTime.substring(9, 12)));
-    return result;
-  }
-
   private void showResults() {
     if (deltaServer.size() < 2 || deltaOCR.size() < 2) {
       Log.d(tag, "No results to show");
       return;
     }
-    Log.d("Show Results", "Video start timestamp:" + CameraFragment.Companion.getRecordingStartTime());
-    String serverChar = "";
-    Integer sync_offset = deltaOCR.get(0) - convertStringTimeToInt(CameraFragment.Companion.getRecordingStartTime());
+    Log.d(
+        "Show Results",
+        "Video start timestamp:" + CameraFragment.Companion.getRecordingStartTime());
+    String serverSequence = "m";
+    sync_offset = deltaServer.get(0) - CameraFragment.Companion.getRecordingStartMillis();
     Log.d("Show Results", "Sync offset:" + sync_offset);
-    for (int i = 1; i < deltaServer.size() && i < deltaOCR.size(); i++) {
-      serverChar+="m";
-      Log.d("Show Results", "i="+ i + "\tServer Text:"+ serverChar + " Time Stamp:"+ deltaServer.get(i) + " (" +  serverCache.get(i) + ")" +
-          "\tOCR text:"+resultsOCR.get(tmpOCR.get(i)) + " Time Stamp:"+ deltaOCR.get(i) +
-          "\tLag="+ (deltaOCR.get(i) - deltaServer.get(i) + sync_offset));
-      analyseResultField.append((deltaOCR.get(i) - deltaServer.get(i) + sync_offset) + "\n");
+    for (int j = 1; j < deltaServer.size(); j++) {
+      for (int i = 0; i < resultsOCR.size(); i++) {
+        if (serverSequence.equals(resultsOCR.get(i))) {
+          analyseResultField.append(
+              "Lag:" + (deltaOCR.get(i) - deltaServer.get(j) + sync_offset)
+                  + " Server sequence:" + serverSequence + "\n");
+          break;
+        }
+      }
+      serverSequence += "m";
```

Comment:
the server text to look for should be sent from the server along with the timestamps, not recalculated in the app.